docs/introduction/index.md

```diff
@@ -0,0 +1,135 @@
+
+If you just want to start scripting right away, importing `shellfish.*` will do the trick; it imports all the extension methods and functionality you need, such as types and functions, to start working right away.
+
+But if you want more concise functionality, the `shellfish.syntax.path.*` and `shellfish.syntax.process.*` will only import the extension methods.
+
+For the static methods, the `shellfish.FilesOs` and `shellfish.ProcessesOs` imports will provide the functions to work with files and processes, if that is your preferred style.
+
+Finally, `shellfish.domain.*` will import all the types needed to work with the library!
+
+## What is this `IO` thing?
+
+You may be wondering at this point, what is this `IO` thing that appears at the end of the functions? Well, that's the [IO monad](https://typelevel.org/cats-effect/docs/2.x/datatypes/io) of Cats Effect; the concept is much more extensive than we can explain on this page, but basically it is a type that allows us to suspend side effects so that they do not run instantly:
+
+```scala mdoc
+import cats.effect.IO
+
+/* Will print to de console! */
+val printingHello = println("Hello newbies!")
+
+/* Will not do anything (yet) */
+val suspendingHello = IO(println("Hello newbies!"))
+```
+
+To actually run the computation, you have two options, the first one (and not recommended) is to call the `unsafeRunSync()` function at the very end of the program:
+
+```scala mdoc
+import cats.effect.unsafe.implicits.global // Imports the runtime that executes the IO monad
+
+suspendingHello.unsafeRunSync()
+```
+
+But this is not the usual way; the common way is to pass it to the `run` function (similar to the main method in scala, but for `IO`!). To that, you have to extend the object with `IOApp`:
+
+```scala mdoc:silent
+import cats.effect.IOApp
+
+object Main extends IOApp.Simple:
+
+  def run: IO[Unit] = suspendingHello
+
+end Main
+```
+
+On either way, the `IO` will be executed and all the computation will be evaluated.
+
+But why's that useful? Well, one of the advantages is referential transparency, and that basically means that we can replace the code wherever it is referenced and expect the same results every time:
+
+```scala mdoc
+val num = 2
+
+(num + num) == (2 + 2)
+```
+
+It may seem trivial, but that's not always the case:
+
+```scala mdoc
+import scala.util.Random
+
+val rndNum = Random.nextInt(10)
+
+val result1 = rndNum + rndNum
+```
+
+If referential transparency exists in your program, replacing `Random.nextInt(10)` in `rndNum` would yield the same result, which is not the case:
+
+```scala mdoc
+val result2 = Random.nextInt(10) + Random.nextInt(10)
+
+result1 == result2
+```
+
+With `IO`, we can solve this by delaying the number generation:
+
+```scala mdoc:compile-only
+import cats.syntax.all.*
+
+val rndNumIO = IO(Random.nextInt(10))
+
+val result1IO = (rndNumIO, rndNumIO).mapN(_ + _)
+
+val result2IO = (IO(Random.nextInt(10)), IO(Random.nextInt(10))).mapN(_ + _)
+```
```

Comment:
Can I ask you to rework this and not use `Random` as an example? You're correctly claiming that `result1 != result2` in the first example, but you are not claiming that the net result of running the second is ``result1 == result2`, because you've picked `Random` that is random and as such would be unpredicable.

Also, don't use `mapN` it will be confusing for newcomers, use a for comprehension.

Use maybe `val salute = println("Hello")` to prove that `salute + salute` won't fire two times.

Reply:
Yup! Let me see what can I do :D